frontend/packages/ceph-storage-plugin/src/components/modals/storage-class-dropdown.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Firehose } from '@console/internal/components/utils';
+import { StorageClassDropdownInner } from '@console/internal/components/utils/storage-class-dropdown';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+
+const cephStorageProvisioners = ['ceph.rook.io/block', 'cephfs.csi.ceph.com', 'rbd.csi.ceph.com'];
+
+export const OCSStorageClassDropdown: React.FC<OCSStorageClassDropdownProps> = (props) => (
+  <Firehose resources={[{ kind: 'StorageClass', prop: 'StorageClass', isList: true }]}>
+    <StorageClassDropdown {...props} />
+  </Firehose>
+);
+
+const StorageClassDropdown = (props) => {
+  const getFilteredStorageClassData = (storageClassData: K8sResourceKind[]): K8sResourceKind[] =>
+    storageClassData.filter(
+      (sc: K8sResourceKind) =>
+        !(
+          _.get(sc, 'provisioner').includes(cephStorageProvisioners[0]) ||
+          _.get(sc, 'provisioner').includes(cephStorageProvisioners[1]) ||
+          _.get(sc, 'provisioner').includes(cephStorageProvisioners[2])
```

Comment:
This is doing a substring match, so you could have false positives. Use `===` if you want to match strings exactly. Assuming you want to keep the ceph providers, use

```js
const cephStorageClasses = storageClassData.filter((sc: K8sRseourceKind) => {
  return cephStorageProvisioners.some((provisioner: string) => sc.provisioner === provisioner);
});
```

Reply:
ack